vrrpd/vrrp.c

```diff
@@ -212,6 +212,15 @@ static struct vrrp_vrouter *vrrp_lookup_by_if_mvl(struct interface *mvl_ifp)
 	}
 
 	p = if_lookup_by_index(mvl_ifp->link_ifindex, VRF_DEFAULT);
+
+	if (!p) {
+		DEBUGD(&vrrp_dbg_zebra,
+		       VRRP_LOGPFX
+		       "Tried to lookup interface %d, parent of %s, but it doesn't exist",
+		       mvl_ifp->link_ifindex, mvl_ifp->name);
+		return NULL;
+	}
```

Comment:
isnot it an issue not finding the interface?

Reply:
If this lookup fails then we simply don't start the router, I don't see the issue.

Reply:
so today, it is not possible to put that interface in a vrf that is not default one ?
ok.

Reply:
I don't understand your comment @pguibert6WIND

Reply:
there should be some comments on the guideline  mentioning that the link_ifindex interface used by macvlan interface should be in default VRF. - this is true for both vrf netns and vrf lite -. the case where macvlan is linked in other namespaces is out of scope of the implementation.

Reply:
But this isn't the case, we do a lookup in the same VRF as the macvlan device.

Reply:
if you move the macvlan device in a VRF, then the algo will fail.
similarly, with netns vrf backend, if you create macvlan in non default netns, then the algo will fail too.
this is what I want to mention as restriction in the guideline.